x/ecocredit/server/expiration.go

```diff
@@ -0,0 +1,51 @@
+package server
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/regen-network/regen-ledger/orm"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+// PruneOrders checks if there are any expired sell or buy orders and removes them from state.
+func (s serverImpl) PruneOrders(ctx sdk.Context) error {
+	blockTime := uint64(ctx.BlockTime().UnixNano())
+	minTime := uint64(0)
+
+	sellOrdersIter, err := s.sellOrderByExpirationIndex.PrefixScan(ctx, minTime, blockTime)
+	if err != nil {
+		return err
+	}
+
+	var sellOrders []*ecocredit.SellOrder
+	_, err = orm.ReadAll(sellOrdersIter, &sellOrders)
```

Comment:
we only need to read the id, not the whole object.

Reply:
True but I'm not sure we have a way to only read the id. As far as I can see, we have to load the whole object here but maybe I'm missing something?

Reply:
Switched to `LoadNext` to consolidate within the loop as suggested but still loading the whole object. Let me know if I'm missing something that would allow reading the id only.

Reply:
`LoadNext` returns `RowID`. I think this is the  ID you are looking for. So we can save empty bytes as the value.

Reply:
Correct but `LoadNext` still loads the whole object. We have to pass a proto message as the destination. `RowID` returns the id in bytes, which we can convert, but we already have the id from the loaded object.